src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    public static class Node<E> {
+        private E item;
+        private Node<E> prev;
+        private Node<E> next;
+
+        public Node(Node<E> prev,E item, Node<E> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> firstNode = new Node<>(null, value,null);
+            tail = firstNode;
+            head = firstNode;
+            size++;
+        } else {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
```suggestion
    public void add(T value, int index) {
            if (index == size) {
            add(value);
            return;
        }
        checkIndexForAddMethod(index);
        Node<T> next = searchNodeByIndex(index);
        Node<T> prev = next.prev;
        Node<T> newNode = new Node<>(prev, value, next);
        if (prev == null) {
            first = newNode;
        } else {
            prev.next = newNode;
        }
        next.prev = newNode;
        size++;
    }
```

we could make it simpler